scene/2d/tile_map_layer.cpp

```diff
@@ -1727,16 +1715,41 @@ void TileMapLayer::_notification(int p_what) {
 void TileMapLayer::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_cell", "coords", "source_id", "atlas_coords", "alternative_tile"), &TileMapLayer::set_cell, DEFVAL(TileSet::INVALID_SOURCE), DEFVAL(TileSetSource::INVALID_ATLAS_COORDS), DEFVAL(0));
 
+	GDVIRTUAL_BIND(_use_tile_data_runtime_update, "coords");
+	GDVIRTUAL_BIND(_tile_data_runtime_update, "coords", "tile_data");
+
 	ADD_SIGNAL(MethodInfo(CoreStringNames::get_singleton()->changed));
 }
 
-void TileMapLayer::set_layer_index_in_tile_map_node(int p_index) {
-	if (p_index == layer_index_in_tile_map_node) {
-		return;
+void TileMapLayer::set_as_tile_map_internal_node(int p_index) {
+	// Compatibility with TileMap.
+	ERR_FAIL_NULL(get_parent());
+	tile_map_node = Object::cast_to<TileMap>(get_parent());
+	set_use_parent_material(true);
+	force_parent_owned();
+	if (layer_index_in_tile_map_node != p_index) {
+		layer_index_in_tile_map_node = p_index;
+		if (layer_index_in_tile_map_node == 0) {
+			// Use main World2D map
+			set_navigation_map(RID());
+		} else {
+			NavigationServer2D *ns = NavigationServer2D::get_singleton();
+			// Check if the map was created manually.
+			RID old_navigation_map = get_navigation_map();
+			if (old_navigation_map.is_valid() && old_navigation_map != get_world_2d()->get_navigation_map()) {
```

Comment:
This isn't guaranteed to be safe, `get_world_2d` can be null unless we're in the tree, and this can be called from outside the tree